torch/utils/launch.py

```diff
@@ -0,0 +1,805 @@
+                                        "\n2. multi-instance \n"
+                                        "\n    >>> python -m intel_extension_for_pytorch.cpu.launch --ninstances xxx --ncore_per_instance xx python_script args\n"
+                                        "\n3. Single-Node multi-process distributed training\n"
+                                        "\n    >>> python  -m intel_extension_for_pytorch.cpu.launch --distributed  python_script args\n"
+                                        "\n4. Multi-Node multi-process distributed training: (e.g. two nodes)\n"
+                                        "\n   rank 0: *(IP: 192.168.10.10, and has a free port: 295000)*\n"
+                                        "\n   >>> python -m intel_extension_for_pytorch.cpu.launch --distributed --nproc_per_node=2\n"
+                                        "\n       --nnodes=2 --hostfile hostfile python_script args\n"
+                                        "\n############################################################################# \n",
+                                        formatter_class=RawTextHelpFormatter)
+
+    parser.add_argument("--multi_instance", action='store_true', default=False,
+                        help="Enable multi-instance, by default one instance per node")
+
+    parser.add_argument('--distributed', action='store_true', default=False,
+                        help='Enable distributed training.')
+    parser.add_argument("-m", "--module", default=False, action="store_true",
+                        help="Changes each process to interpret the launch script "
+                             "as a python module, executing with the same behavior as"
+                             "'python -m'.")
+
+    parser.add_argument("--no_python", default=False, action="store_true",
+                        help="Do not prepend the --program script with \"python\" - just exec "
+                             "it directly. Useful when the script is not a Python script.")
+
+    add_memory_allocator_params(parser)
+    add_kmp_iomp_params(parser)
+
+    add_distributed_training_params(parser)
+    add_multi_instance_params(parser)
+    # positional
+    parser.add_argument("program", type=str,
+                        help="The full path to the proram/script to be launched. "
+                             "followed by all the arguments for the script")
+
+    # rest from the training program
+    parser.add_argument('program_args', nargs=REMAINDER)
+    return parser.parse_args()
+
+def main():
+
+    env_before = set(os.environ.keys())
+    if platform.system() == "Windows":
+        raise RuntimeError("Windows platform is not supported!!!")
+
+    args = parse_args()
+    if args.log_path:
+        path = os.path.dirname(args.log_path if args.log_path.endswith('/') else args.log_path + '/')
+        if not os.path.exists(path):
+            os.makedirs(path)
```

Comment:
Please use `exist_ok` flag, see https://docs.python.org/3.7/library/os.html#os.makedirs
```suggestion
        os.makedirs(path, exist_ok=True)
```

Reply:
Sure.